### PR TITLE
bundle update rubocop --conservative

### PR DESCRIPTION
**Pull request by skipkayhil**
- fixes a new case of [`Style/RedundantReturn`][1]
- fixes a new case of [`Performance/StringInclude`][2]

[1]: https://github.com/skipkayhil/rails/commit/146b1c2e3389bc70ea0b54abf7843fc1d6c8cd5f
[2]: https://github.com/skipkayhil/rails/commit/3158bbb9f6454dce64dd0b4e2a548351d014c48f

---

When attempting to update Gemfile.lock for Ruby 3.3.0 (#50516), there were a few new warnings raised by RuboCop. This commit splits out just the RuboCop update with fixes for the new warnings introduced.

### CODE REVIEW

File: actionpack/test/controller/routing_test.rb [hunk 1]
```diff
@@ -1694,7 +1694,7 @@ def test_route_with_subdomain_and_constraints_must_receive_params
     set.draw do
       get "page/:name" => "pages#show", :constraints => lambda { |request|
         name_param = request.params[:name]
-        return true
+        true
```
Comment (dhh) [inline actionpack/test/controller/routing_test.rb:1697]: Don't like this rule.

File: actionview/lib/action_view/helpers/url_helper.rb [hunk 2]
```diff
@@ -784,7 +784,7 @@ def link_to_remote_options?(options)
         end
 
         def add_method_to_attributes!(html_options, method)
-          if method_not_get_method?(method) && !html_options["rel"]&.match?(/nofollow/)
+          if method_not_get_method?(method) && !html_options["rel"]&.include?("nofollow")
```
Comment (dhh) [inline actionview/lib/action_view/helpers/url_helper.rb:787]: This also seems like an odd choice. Maybe fine but it's in the wrong PR.

File: actionpack/test/controller/routing_test.rb (hunk 1 above)
Comment (skipkayhil) [reply actionpack/test/controller/routing_test.rb:1697]: (linking for context) It looks like the rule was enabled in 146b1c2e3389bc70ea0b54abf7843fc1d6c8cd5f after #31004 was merged

File: actionview/lib/action_view/helpers/url_helper.rb (hunk 2 above)
Comment (skipkayhil) [reply actionview/lib/action_view/helpers/url_helper.rb:787]: (linking for context) `Performance/StringInclude` was enabled in #45865

File: actionpack/test/controller/routing_test.rb (hunk 1 above)
Comment (dhh) [reply actionpack/test/controller/routing_test.rb:1697]: Guess that's why I'd be keen to get Rails under omakase style too.

**Comment by skipkayhil** [comment]
> Maybe fine but it's in the wrong PR.

I'm not sure I understand the reasoning here. The rules are already being enforced but bumping RuboCop is bringing in bug fixes that cause these two code changes to only be warned on now.

File: actionview/lib/action_view/helpers/url_helper.rb (hunk 2 above)
Comment (dhh) [reply actionview/lib/action_view/helpers/url_helper.rb:787]: Gotcha. Not sure about this one either! But yes, should be in compliance with the rules as they currently are. Just think that needs to happen in a clear PR separate from a dependency bump.

**Comment by dhh** [comment]
The dependency bump is broad, bumps all sorts of stuff. The new rubocop rules are specific, and should just get their own explaining PR. Or you can split the dep bump into "just rubocop, plus new rules" and "everything else".

**Comment by skipkayhil** [comment]
> The dependency bump is broad, bumps all sorts of stuff. The new rubocop rules are specific, and should just get their own explaining PR. Or you can split the dep bump into "just rubocop, plus new rules" and "everything else".

Got it, the transitive dependencies are being bumped as well but really we should bump the minimum to trigger these violations to fix. I'll make that change 👍

**Pull request merged**